src/Agent/NewRelic/Agent/Core/DataTransport/Client/HttpClientBase.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using NewRelic.Agent.Core.DataTransport.Client.Interfaces;
+using NewRelic.Core.Logging;
+
+namespace NewRelic.Agent.Core.DataTransport.Client
+{
+    /// <summary>
+    /// Abstract base shared by implementations of IHttpClient
+    /// </summary>
+    public abstract class HttpClientBase : IHttpClient
+    {
+        protected bool _diagnoseConnectionError = true;
+        protected static IWebProxy _proxy = null;
+
+        protected HttpClientBase(IWebProxy proxy)
+        {
+            _proxy = proxy;
+        }
+
+        public abstract Task<IHttpResponse> SendAsync(IHttpRequest request);
+        public abstract void Dispose();
+
+
+        protected void DiagnoseConnectionError(string host)
+        {
+            _diagnoseConnectionError = false;
+            try
+            {
+                if (!IPAddress.TryParse(host, out _))
+                {
+                    Dns.GetHostEntry(host);
+                }
+            }
+            catch (Exception)
+            {
+                Log.ErrorFormat("Unable to resolve host name \"{0}\"", host);
+            }
+
+            TestConnection();
+        }
+
+#if NETFRAMEWORK
+        // use WebClient for .NETFramework builds
```

Comment:
Nitpick: Maybe we could combine these and have the `#if` inside the call? It's really just the Get that's different